developer/build_plugin.py

```diff
@@ -495,11 +655,29 @@ def main():
 
     # now authenticate to shotgun
     sg_auth = ShotgunAuthenticator()
-    sg_user = sg_auth.get_user()
+
+    if options.shotgun_host:
+        script_name = options.shotgun_script_name
+        script_key = options.shotgun_script_key
+
+        if script_name is None or script_key is None:
+            logger.error("Need to provide, host, script name and script key! Run with -h for more info.")
+            return
+
+        logger.info("Connecting to %s using script user %s..." % (options.shotgun_host, script_name))
+        sg_user = sg_auth.create_script_user(script_name, script_key, options.shotgun_host)
+
+    else:
+        # get user, prompt if necessary
+        sg_user = sg_auth.get_user()
+
     sg_connection = sg_user.create_sg_connection()
+    # make sure we are properly connected
+
+    sg_connection.find_one("HumanUser", [])
```

Comment:
If this is testing the connection, maybe it should be wrapped in a `try/except`?